app/src/main/java/net/squanchy/support/graphics/CircularRevealDrawable.kt

```diff
@@ -0,0 +1,125 @@
+package net.squanchy.support.graphics
+
+import android.animation.Animator
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.Rect
+import android.graphics.drawable.ColorDrawable
+import android.support.annotation.ColorInt
+import android.support.annotation.IntRange
+import android.support.v4.view.animation.FastOutLinearInInterpolator
+import me.eugeniomarletti.renderthread.CanvasProperty
+import me.eugeniomarletti.renderthread.RenderThread
+import kotlin.math.max
+import kotlin.math.sqrt
+
+class CircularRevealDrawable : ColorDrawable() {
+
+    private val animationPaint = Paint(Paint.ANTI_ALIAS_FLAG)
+
+    private val interpolator = FastOutLinearInInterpolator()
+
+    private lateinit var radiusAnimator: Animator
+
+    private lateinit var centerXProperty: CanvasProperty<Float>
+    private lateinit var centerYProperty: CanvasProperty<Float>
+    private lateinit var radiusProperty: CanvasProperty<Float>
+    private lateinit var paintProperty: CanvasProperty<Paint>
+
+    private var startAnimationOnNextDraw: Boolean = false
+
+    private var revealDuration: Int = 0
+
+    private var hotspotX: Float = 0f
+    private var hotspotY: Float = 0f
+    private var width: Int = 0
+    private var height: Int = 0
+
+    @ColorInt
+    private var pendingTargetColor: Int = 0
+
+    @ColorInt
+    private var targetColor: Int = 0
+
+    fun animateToColor(@ColorInt newColor: Int, @IntRange(from = 0) durationMillis: Int) {
+        revealDuration = durationMillis
+        pendingTargetColor = newColor
+        startAnimationOnNextDraw = true
+        color = targetColor
+    }
+
+    override fun draw(canvas: Canvas) {
```

Comment:
can we extract meaningfully named methods for the lines that have comments since we're at it?